docs/miscellaneous.rst

```diff
@@ -31,6 +31,14 @@ Statically-sized variables (everything except mapping and dynamically-sized arra
     declaring your storage variables in the order of ``uint128, uint128, uint256`` instead of
     ``uint128, uint256, uint128``, as the former will only take up two slots of storage whereas the
     latter will take up three.
+ 
+ .. note::
+     The layout of state variables in storage is considered to be part of the binary API
```

Comment:
Isn't a ``binary API`` an ``ABI`` :-)?

Reply:
You are right, but I'm not sure if that would confuse people, since ABI unfortunately has a very specific meaning in the Ethereum ecosystem.

Reply:
Yes - I'm not sure whether ``binary API`` will be any less confusing, though... do we actually need to name this here? It would probably be enough to just say that changing the layout would break the interface with libraries and to continue with the second sentence from there.